From 4ba2268443c7c00cef27a4ebe11dea15f6e932e4 Mon Sep 17 00:00:00 2001
From: Shreenidhi Shedi <sshedi@vmware.com>
Date: Thu, 3 Nov 2022 18:02:10 +0530
Subject: [PATCH] Makefile: Set VERSION & REVISION only if unset

This is useful while building the package from git tag.
For example, while building rpm from a tagged source tarball.

Signed-off-by: Shreenidhi Shedi <sshedi@vmware.com>
---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index af2fbf9..d1b8c5c 100644
--- a/Makefile
+++ b/Makefile
@@ -14,9 +14,9 @@
 
 DESTDIR ?= /usr/local
 
-VERSION=$(shell git describe --match 'v[0-9]*' --dirty='.m' --always --tags)
+VERSION ?= $(shell git describe --match 'v[0-9]*' --dirty='.m' --always --tags)
 VERSION_TRIMMED := $(VERSION:v%=%)
-REVISION=$(shell git rev-parse HEAD)$(shell if ! git diff --no-ext-diff --quiet --exit-code; then echo .m; fi)
+REVISION ?= $(shell git rev-parse HEAD)$(shell if ! git diff --no-ext-diff --quiet --exit-code; then echo .m; fi)
 
 PKG_MAIN := github.com/containerd/fuse-overlayfs-snapshotter/cmd/containerd-fuse-overlayfs-grpc
 PKG_VERSION := github.com/containerd/fuse-overlayfs-snapshotter/cmd/containerd-fuse-overlayfs-grpc/version
-- 
2.35.5

